### DISCUSSION TITLE
V1 Breaking Changes Signal Inputs

### DISCUSSION THREAD

**Discussion by elite-benni** [spartan-ng/spartan#456] [General]
I am quoting @ashley-hunter here from the PR . 
https://github.com/spartan-ng/spartan/pull/445#discussion_r1834341341

```
Thank you!

Yes, so we have identified the list of issues we want to get completed as part of the lead up to our v1 release, and we have identified signal migration as a place where breaking changes are likely to occur, and this has proven to be the case in the few PRs for signal migrations that have arisen.

In my opinion we are still in alpha, so now is the time for us to get the codebase in the best shape before v1 as breaking changes after that point are a much bigger issue. Having unneeded aliasing and other workarounds etc, is going to clutter the code base and make it less maintainable going forward.

We have seen similar cases in https://github.com/spartan-ng/spartan/pull/447 and https://github.com/spartan-ng/spartan/pull/449, and I suspect there will be more too.

That being said, breaking changes are not ideal, and we should minimize there impact on consumers. I do think it could be worth having an upgrade guide detailing all the breaking changes and showing the fix, in this case, it would be a rename, but in the cases like button, it may be slightly more involved. We have a Changelog page on the docs which is one place where this kind of content could be put, but I'm open to suggestions.

Would be keen to hear your thoughts on how we handle these breaking changes that are going to arise @thatsamsonkid, @goetzrobin, @elite-benni. I think we should come to a consensus so we know how to approach this going forward.

Do we make the breaking changes now to improve the codebase while we are still in alpha?
If so should we document it? And where?
```

**Comment by elite-benni** [comment]
I would say this is the purpose of Alpha and as I understood it, writing to the input was not intended by the angular team. 

Technically we are followig best practices here. 
I would say we make the Breaking changes now, because i think it is the future and it has to be changed anyways someday.

Maybe we can write a Infopage where we inform that we are working towards V1 and that These are the changes that needs to be made.
We can link this page in the alpha warning banner.

**Comment by ashley-hunter** [comment]
> I would say this is the purpose of Alpha and as I understood it, writing to the input was not intended by the angular team.

Exactly, while this technically worked before, it wasn't the intention. Signal inputs enforce a much better practice ensuring that there is only a single source of truth.

> Maybe we can write a Infopage where we inform that we are working towards V1 and that These are the changes that needs to be made. We can link this page in the alpha warning banner.

Sounds like a good idea! I know quite a few people are curious about when 1.0 will arrive, so it would be good to have some info somewhere about our recent progress and plan to get there.

**Reply by goetzrobin** [reply]
I also talked to Aleš about a blog post announcing his support. I think blog post and dedicated page as part of the documentation might be a way to make sure everyone is up to date. Thoughts?

**Reply by ashley-hunter** [reply]
Sounds like a good idea to me! We will also likely want a blog post announcing the 1.0 release when we get there, so a general "blog" section could be a useful area to have on the site? I can foresee other potential posts about zerops templates etc in future?

**Reply by thatsamsonkid** [reply]
I would also just recommend throwing something in the discord general chat as well and we can point to the blog post. For this I would rather overcommunicate then have people say we didnt do enough to inform them of these changes. Even though we are in alpha.
